### ISSUE TITLE
Autoload config from AppData

### ISSUE THREAD

**Issue by QuartzSTQ** [chocolate-doom/chocolate-doom#1204]
### Background

Version of Chocolate Doom: latest build from [here](http://latest.chocolate-doom.org/) (also applies to crispy-doom)

Operating System and version: Windows 10 (should also apply to any NT version of Windows, presumably)

### Feature description

Anyways, this is exactly what the title suggests:
If config files are found in the %APPDATA%\chocolate-doom, load those instead of ones found in the current directory. I'm not saying don't use them, in fact, those should still be used by default. I'm just saying that if config files **ARE** found, use them.

I did find another problem in that I got the idea in the from first place:
Because in recent builds of chocolate-doom (Not sure of the exact commit) it defaults to checking AppData for music packs and PWADs that should be automatically loaded. And guess what... You can specify exactly where to search for those... In the config. I feel like doing the same for those as I suggested for config files would be the solution here (Search for them in AppData, and if not found use those in the current directory).

**Comment by QuartzSTQ** [comment]
Right, I forgot to mention the use case for this, and while it may seem a bit niche, I still wanna try it out, anyways, here goes:
A centralized directory containing only executables, that is added to PATH.

I also thought out a temporary solution if editing the software itself is out of bounds.
Use three (or two, that should still work now that I think about it) directories:

- One (That is added to PATH) to store executables that don't create any other files in the current directory
- Another (Isn't added to PATH) to store executables that do require files created in the same folder unless specified by an argument (chocolate-doom and -config)
- And the final (Added to PATH) to store simple shell scripts to make the executables from the second folder work (Also this directory doesn't really need to be separated from the first directory)

I feel like this solution kinda ruins the whole ideology of having one simple folder for everything executable though.
Also now that I think about it, it doesn't need to be necesarily AppData, and you could specify it using,  say, another enviroment variable... Problem is AppData kinda is that already, so yeah.

**Comment by fragglet** [comment]
I like the idea. There's a catch - we already create this directory, for autoloading and for music packs. So it specifically needs to be "use the configuration files if they exist there" and not "generate configuration files there if the directory exists".

Note that on all other platforms other than Windows we already use a central configuration location. It's only on Windows that we do it per-folder, because it descends from MS-DOS and we're emulating Vanilla Doom's behavior. Also because of how we distribute the executables.
